awslimitchecker/services/dynamodb.py

```diff
@@ -0,0 +1,246 @@
+        table_count = 0
+        region_read_capacity = 0
+        region_write_capacity = 0
+        dynamodb_client = self.conn
+
+        for table in dynamodb_client.list_tables()['TableNames']:
+            table_count += 1
+            gsi_write = 0
+            gsi_read = 0
+
+            table_desc = dynamodb_client.describe_table(TableName=table)['Table']
+
+            logger.debug("Getting usage for DynamoDB tables")
+
+            if 'GlobalSecondaryIndexes' in table_desc:
+                for gsi in table_desc['GlobalSecondaryIndexes']:
+                    gsi_read += gsi['ProvisionedThroughput']['ReadCapacityUnits']
+                    gsi_write += gsi['ProvisionedThroughput']['WriteCapacityUnits']
+
+            table_write_capacity = table_desc['ProvisionedThroughput']['WriteCapacityUnits'] + gsi_write
+            table_read_capacity = table_desc['ProvisionedThroughput']['ReadCapacityUnits'] + gsi_read
+            region_write_capacity += table_write_capacity
+            region_read_capacity += table_read_capacity
+
+            self.limits['Global Secondary Indexes'
+            ]._add_current_usage(
+                len(table_desc['GlobalSecondaryIndexes']
+                    if 'GlobalSecondaryIndexes' in table_desc else 0),
+                resource_id=table_desc['TableArn'],
+                aws_type='AWS::DynamoDB::Table'
+            )
+
+            self.limits['Local Secondary Indexes'
+            ]._add_current_usage(
+                len(table_desc['LocalSecondaryIndexes']
+                    if 'LocalSecondaryIndexes' in table_desc else 0),
+                resource_id=table_desc['TableArn'],
+                aws_type='AWS::DynamoDB::Table'
+            )
+
+            self.limits['Write Capacity (table)'
+            ]._add_current_usage(
+                table_write_capacity,
+                resource_id=table_desc['TableArn']
+            )
+
+            self.limits['Read Capacity (table)'
+            ]._add_current_usage(
+                table_read_capacity,
+                resource_id=table_desc['TableArn']
+            )
+
+        self.limits['Table Count (region)'
+        ]._add_current_usage(
+            table_count,
+            aws_type='AWS::DynamoDB::Table'
+        )
+
+        self.limits['Write Capacity (region)'
+        ]._add_current_usage(
+            region_write_capacity,
+            aws_type='AWS::DynamoDB::Table'
+        )
+
+        self.limits['Read Capacity (region)'
+        ]._add_current_usage(
+            region_read_capacity,
+            aws_type='AWS::DynamoDB::Table'
+        )
+
+    def get_limits(self):
+        """
+        Return all known limits for this service, as a dict of their names
+        to :py:class:`~.AwsLimit` objects.
+
+        :returns: dict of limit names to :py:class:`~.AwsLimit` objects
+        :rtype: dict
+        """
+        region_name = self.conn._client_config.region_name
```

Comment:
You need a call to `self.connect()` before this line.

Reply:
that's exactly what I did

Reply:
Ok. Can you then try to post the full traceback, and debug-level output?